snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -257,6 +257,7 @@ def makedirs(self):
         dirs = [
             self.part_source_dir,
             self.part_build_dir,
+            self.part_build_work_dir,
```

Comment:
Why create this here? Isn't the source copied over wholesale (or, as of this PR, potentially not at all)?

Reply:
Still copied but wiped (code block below), and now that I see, wiped every time killing our rebuild story

Reply:
> wiped every time killing our rebuild story

Remember we also have `update_build()`.

Reply:
I needed this change to allow the `cmake-hello-subdir` spread test to pass.  That snap has a part with `source-subdir: subdir` set, which meant that it tries to execute the build script in `parts/hello/build/subdir`.  Without copying the source code over, this directory wouldn't otherwise exist.

Reply:
That is because this snapcraft accounts for the working directory when subdir is used: 
https://github.com/snapcore/snapcraft/blob/master/snapcraft/internal/pluginhandler/__init__.py#L684

Specifically `cwd=self.part_build_work_dir`

If using out of source builds, I think `cwd=self.part_build_dir` is appropriate?

Reply:
I've updated the code to set part_build_work_dir to part_build_dir when a plugin uses out_of_source_build.  That will have the build occur in the base build dir, and make sure `$SNAPCRAFT_PART_BUILD_WORK` actually points at where the build is occurring, if it happens to be used by a scriptlet.